portfolio/src/main/webapp/script.js

```diff
@@ -15,14 +15,24 @@
 /**
  * Adds a random greeting to the page.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
-
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+async function fetchVideoData(id) {
+  fetch('/data?videoId=' + id)  // sends a request to /data URl with a cursor and whether we want next page or not
```

Comment:
NIT: In case you haven't seen it, the await command is nice for chains of promises like this.